scipy/stats/_survival.py

```diff
@@ -374,3 +391,138 @@ def _ecdf_right_censored(sample):
     sf = np.cumprod((n - d) / n)
     cdf = 1 - sf
     return t, cdf, sf, n, d
+
+
+@dataclass
+class LogRankResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+
+
+def log_rank(
+    x: npt.ArrayLike | CensoredData,
+    y: npt.ArrayLike | CensoredData
+) -> LogRankResult:
+    """Compare the survival distributions of two samples via the logrank test.
+
+    Parameters
+    ----------
+    x, y : array_like or CensoredData
+        Samples to compare based on their survival functions.
+
+    Returns
+    -------
+    res : LogRankResult
+        An object containing attributes:
+
+        statistic : float ndarray
+            The computed statistic of the test.
+        pvalue : float ndarray
+            The computed p-value of the test.
+
+    See Also
+    --------
+    scipy.stats.ecdf
+
+    References
+    ----------
+    .. [1] Peto, Richard and Peto, Julian. "Asymptotically Efficient Rank
+           Invariant Test Procedures." J. R. Statist. Soc. A, 185 (1972)
+    .. [2] Bland, Altman, "The logrank test", BMJ, 328:1073,
+           :doi:`10.1136/bmj.328.7447.1073`, 2004
+
+    Examples
+    --------
+    In [2]_, the survival of two different group of patient with recurrent
+    malignant gliomas was investigated. The two groups are characterized by
+    a different type of tumor. The week to death of 51 adults was recorded.
```

Comment:
```suggestion
    a different type of tumor. The number of weeks to death of 51 adults was recorded.
```
Check PEP8 after these suggestions.

Please mention why CensoredData is being used. This is not always the number of weeks to death. It is always the number of weeks until the patient left the study, whether due to death (uncensored) or some other reason (censored).

Reply:
This has not all been addressed, and a typo was introduced, but I'll go ahead and fix that.